### DISCUSSION TITLE
Ban Button in Multiplayer Lobbies

### DISCUSSION THREAD

**Discussion by TRUEchips** [ppy/osu#23219] [Ideas]
So, I was playing with a friend recently, and some rude person just kept joining and adding blank beatmaps even though we didn't want him to keep playing with us. It took like half an hour of constant kicking for him to not even give up! Like what the hell dude, we don't want you, AND your blank troll beatmaps, and we want to keep the lobby open for other reasons, same with the open beatmap queues. 

So, essentially, I BEG you to add a button that bans these kinds of cases, by just - ban - and the guy cannot join that particular lobby ever again. The mod I contacted couldn't unfortunately do anything but advise us to close the lobby, which while understandable, is extremely infuriating, that these people can just get away with this and still bother you even after reporting.

**Comment by peppy** [comment]
Set the lobby size to two?

**Reply by Joehuu** [reply]
That is not a thing in lazer yet.

**Reply by uzervlad** [reply]
So instead of dealing with obnoxious people, you suggest just not making an open lobby?

**Reply by peppy** [reply]
Generally if you're just playing with a friend that's the best way, yeah. There's also a setting to disallow non-hosts to add beatmaps, if you'd prefer to let others play but not add beatmaps.

**Reply by TRUEchips** [reply]
Yeah like I said, not the prefferable choice for me, especially that both me and my friend like to add beatmaps. Additionally, stable already has banning, so I don't see this as being especially hard to implement, and plus plus, adding a button would be just far more convenient

**Reply by ziv-vy** [reply]
You might aswell have said "Stop playing with strangers". Then why do we have public multiplayer lobbies in the first place? If someone picks Centipede three times, gets kicked, immediately rejoins and drops the N word, is it my fault that I didn't set a password and wanted to let strangers join?.

**Comment by Joehuu** [comment]
Doesn't setting a password resolve this?

**Comment by monochrome22** [comment]
The suggested solutions in this thread are too focused on this specific case, instead of the broader issue of having obnoxious people messing with multiplayer lobbies. To me it seems obvious that there should be a ban option, it was even added to stable. 
Would be interested to know why peppy and smoogi (https://github.com/ppy/osu/discussions/21610) are against it. Are you concerned it will be used in negative ways? That can already be done with kick.

**Reply by TRUEchips** [reply]
Indeed. I think so too. Additionally, like you said, a ban command already exists in stable, and a ban button is in multiplayer of pretty much every multiplayer game with public lobbies.
